### PR TITLE
Added missing `BT_ENABLE_APPLE_PAY` check

### PR DESCRIPTION
**Pull request by Reflejo**
Without this, when `BT_ENABLE_APPLE_PAY` is undefined the compiler complains about `BTClientTokenApplePayStatusValueTransformer` not being defined.

### CODE REVIEW

**Comment by burnto** [comment]
Thanks @Reflejo - this is great. I'd like to reduce the scope of the check to within the implementation. See #89. What do you think?

**Comment by Reflejo** [comment]
My motto is "fail early, fail hard". But I understand why you'd want the implementation to still work even when `BT_ENABLE_APPLE_PAY` is not defined and make it behave more like as a "kill switch" for apple pay. The only problem with the "work as normal even when apple pay is not enabled by the compilation flag" approach is that there is no warning or error if one mistakenly forgets to set `-D BT_ENABLE_PAYPAL=1`.

I, for once, don't like the flag checks everywhere because makes the code more complex and (as it depends on a global flag) hard to follow. Also: there is no BT_ENABLE_PAYPAL or BT_ENABLE_VENMO which makes it less consistent but I'm sure you guys have a good reason for adding this on apple pay.

**Comment by Reflejo** [comment]
Having said that, I'm :+1: with the other approach and you can close this PR

**Comment by burnto** [comment]
Thanks @Reflejo. We don't like the `BT_ENABLE_APPLE_PAY` preprocessor macro checks everywhere either. We added this macro and this build configuration option to address #60. This approach was the only option given the way we integrated and exposed Apple Pay in v.zero. We hope to better decouple the various payment methods in the next major version bump.

**Comment by burnto** [comment]
Going with #89 

Thanks @Reflejo! :star2: :sunglasses: :star2:
